test/unit/adb-commands-specs.js

```diff
@@ -0,0 +1,79 @@
+import chai from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+import 'mochawait';
+import ADB from '../../lib/adb.js';
+import sinon from 'sinon';
+
+chai.use(chaiAsPromised);
+const should = chai.should();
+const apiLevel = '21',
+      IME = 'com.android.inputmethod.latin/.LatinIME',
+     imeList = `com.android.inputmethod.latin/.LatinIME:
+  mId=com.android.inputmethod.latin/.LatinIME mSettingsActivityName=com.android
+  mIsDefaultResId=0x7f070000
+  Service:
+    priority=0 preferredOrder=0 match=0x108000 specificIndex=-1 isDefault=false
+    ServiceInfo:
+      name=com.android.inputmethod.latin.LatinIME
+      packageName=com.android.inputmethod.latin
+      labelRes=0x7f0a0037 nonLocalizedLabel=null icon=0x0 banner=0x0
+      enabled=true exported=true processName=com.android.inputmethod.latin
+      permission=android.permission.BIND_INPUT_METHOD
+      flags=0x0`;
+
+describe('adb commands', () => {
+  let adb = new ADB(), shell;
+  const createStub = function (mockShellValue) {
+    shell = sinon.stub(adb, "shell", async function () {
+      return mockShellValue;
+    });
+  };
+  beforeEach(async () => {
+    await adb.createADB();
+  });
+  describe('shell', () => {
+    afterEach(async () => {
+      adb.shell.restore();
+    });
+    let verifyShellArguments = function (cmd) {
```

Comment:
sinon does not allow me to mock already stubbed function 

`TypeError: Attempted to wrap shell which is already wrapped`